float64 x
float64 y
float64 z
float64 x_dot
float64 y_dot
float64 z_dot
float64 phi
float64 theta
float64 psi
float64 p
float64 q
float64 r

